### PR TITLE
fix: support array of references for IN queries

### PR DESCRIPTION
**Pull request by thebrianchen**
Porting from [node](https://github.com/googleapis/nodejs-firestore/pull/993).

### CODE REVIEW

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java [hunk 1]
```diff
@@ -709,7 +711,27 @@ public Query whereIn(@Nonnull FieldPath fieldPath, @Nonnull List<? extends Objec
...
       FieldPath fieldPath, StructuredQuery.FieldFilter.Operator operator, Object value) {
     if (fieldPath.equals(FieldPath.DOCUMENT_ID)) {
-      value = this.convertReference(value);
+      if (operator == ARRAY_CONTAINS || operator == ARRAY_CONTAINS_ANY) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Invalid query. You cannot perform '%s' queries on FieldPath.documentId().",
+                operator.toString()));
+      } else if (operator == IN) {
+        if (!(value instanceof List) || ((List<?>) value).size() == 0) {
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java:720]: Nit: Prefer `isEmpty()` as it is guaranteed to run in `O(1)`.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java [hunk 2]
```diff
@@ -273,6 +274,87 @@ public void withFieldPathFilter() throws Exception {
...
+    } catch (IllegalArgumentException e) {
+      assertEquals(
+          "The corresponding value for FieldPath.documentId() must be a String or a "
+              + "DocumentReference, but was: 42.",
+          e.getMessage());
+    }
+
+    try {
+      query.whereIn(FieldPath.documentId(), Arrays.<Object>asList(42)).get();
+      fail();
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java:320]: The Node tests uses "42", not "Arrays.<Object>asList(42)", which tests a different case. I think you are re-testing the case from right above here.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java (hunk 1 above)
Comment (thebrianchen) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java]: done.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java (hunk 2 above)
Comment (thebrianchen) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java]: Removed test, since passing in "42" causes compiler error.

**Pull request merged**